microcosm_postgres/encryption/registry.py

```diff
@@ -11,23 +11,32 @@
 
 from microcosm_postgres.encryption.encryptor import MultiTenantEncryptor, SingleTenantEncryptor
 from microcosm_postgres.encryption.providers import (
-    configure_key_provider,
+    configure_decrypting_key_provider,
+    configure_encrypting_key_provider,
     configure_materials_manager,
 )
 
 
 def parse_config(context_keys: Sequence[str],
-                 key_ids: Sequence[str]) -> Mapping[str, Sequence[str]]:
+                 key_ids: Sequence[str],
+                 account_ids: Sequence[str],
+                 partitions: Sequence[str]) -> Mapping[str, Mapping[str, Sequence[str]]]:
     return {
         # NB: split key id on non-comma to avoid confusion with config parsing
-        context_key: key_id.split(";") if isinstance(key_id, str) else key_id
-        for context_key, key_id in zip(context_keys, key_ids)
+        context_key: {
+            "key_ids": key_id.split(";") if isinstance(key_id, str) else key_id,
+            "account_ids": account_id.split(";") if isinstance(account_id, str) else account_id,
```

Comment:
`key_id` and `account_id` look like they should always be `str` according to the type hint – do we know which is correct here?

Reply:
Hm, this may have been a compromise stemming from its usage in unit tests (perhaps a kind of split-brain between configuring purely in python and configuring from envvars?)

Regardless, I can update to make it firmly decide, or just make it a union.

Reply:
From what it looks like, one should be able to deal entirely with one of either envvar config, or in-app config, in which case both are valid. I've updated the typehints accordingly.